# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ContentParent.cpp
# Commit: 06c26e973285
# Full Hash: 06c26e97328589418681f58d637df751451f04ca
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2020-05-26 21:37:52
# Regressor Bug: 1640119
# File Overlap Count: 1
# Description:
#   Bug 1640119 - Pass shared-memory blocks for the font list as part of SetXPCOMProcessAttributes. r=jwatt
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D76507
# ==============================================================================

diff -r fbdbe0cdce92 -r 06c26e973285 dom/ipc/ContentParent.cpp
--- a/dom/ipc/ContentParent.cpp	Tue May 26 11:38:58 2020 +0000
+++ b/dom/ipc/ContentParent.cpp	Tue May 26 09:30:17 2020 +0000
@@ -2358,6 +2358,12 @@
   gfxPlatform::GetPlatform()->ReadSystemFontList(&fontList);
   nsTArray<LookAndFeelInt> lnfCache = LookAndFeel::GetIntCache();
 
+  // If the shared fontlist is in use, collect its shmem block handles to pass
+  // to the child.
+  nsTArray<SharedMemoryHandle> sharedFontListBlocks;
+  gfxPlatformFontList::PlatformFontList()->ShareFontListToProcess(
+      &sharedFontListBlocks, OtherPid());
+
   // Content processes have no permission to access profile directory, so we
   // send the file URL instead.
   auto* sheetCache = GlobalStyleSheetCache::Singleton();
@@ -2401,9 +2407,9 @@
     sharedUASheetAddress = 0;
   }
 
-  Unused << SendSetXPCOMProcessAttributes(xpcomInit, initialData, lnfCache,
-                                          fontList, sharedUASheetHandle,
-                                          sharedUASheetAddress);
+  Unused << SendSetXPCOMProcessAttributes(
+      xpcomInit, initialData, lnfCache, fontList, sharedUASheetHandle,
+      sharedUASheetAddress, sharedFontListBlocks);
 
   ipc::WritableSharedMap* sharedData =
       nsFrameMessageManager::sParentProcessManager->SharedData();